### PR TITLE
Allow beans to be injected in tests even if they aren't used in the application

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #1749

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/test/TestDefiningAnnotationBuildItemProvider.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package io.quarkus.deployment.test;
+
+import org.jboss.builder.BuildStep;
+import org.jboss.builder.item.BuildItem;
+import org.jboss.jandex.DotName;
+
+// this is only needed because the test modules do not depend on Arc
+public interface TestDefiningAnnotationBuildItemProvider {
```
Comment (stuartwdouglas) [inline core/deployment/src/main/java/io/quarkus/deployment/test/TestDefiningAnnotationBuildItemProvider.java]: I don't know if we really need this SPI, as it is not really usable outside this specific case.

We could probably just do:


----
    @BuildStep
    BeanDefiningAnnotationBuildItem testBeans(LaunchModeBuildItem launchModeBuildItem) {
        if(launchModeBuildItem.getLaunchMode() == LaunchMode.TEST) {
            return new BeanDefiningAnnotationBuildItem(DotName.createSimple("io.quarkus.test.junit.QuarkusTest"));
        }
        return null;
    }
----

For Junit 4 you could also declare @RunWith as a bean defining annotation.

In some ways its a bit less 'pure' as you have the test annotation names in the ArC module however it is a lot simpler and should be a lot more obvious what is going on.

**Review by stuartwdouglas — DISMISSED** [review]

File: core/deployment/src/main/java/io/quarkus/deployment/test/TestDefiningAnnotationBuildItemProvider.java (hunk 1 above)
Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/test/TestDefiningAnnotationBuildItemProvider.java]: Sounds good to me! I will go ahead and make the changes you propose.

**Review by mkouba — APPROVED** [review]
Looks good. In theory, if we want a more correct solution we could create a `TestAnnotationBuildItem` in the core-deployment, and add another build step [here](https://github.com/quarkusio/quarkus/blob/master/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java#L232)  and then the `TestsAsBeanDefiningAnnotationsProcessor` would consume this build item.

**Comment by geoand** [comment]
@mkouba do you think we should make that change now or in the future if we have more changes to do around testing and Arc?

**Comment by mkouba** [comment]
@geoand It's up to you ;-). The change should be pretty straightforward. If we decide to go with hard-coded values we should at least add some comment about the possible improvement.

**Comment by geoand** [comment]
@mkouba Let me think about for a few minutes :P. The change is indeed super easy to make and I the reason I think it's good to do is to keep things consistent with what we have done so far

**Comment by geoand** [comment]
@mkouba PR updated :wink:

**Comment by mkouba** [comment]
Thanks, looks good! Let's wait for CI.

**Pull request merged**